allennlp/data/dataset_readers/huggingface_datasets_reader.py

```diff
@@ -0,0 +1,270 @@
+                fields[field_key] = fields_to_be_added[field_key]
+
+        return Instance(fields)
+
+
+# Feature Mappers - These functions map a FeatureType into Fields
+def map_Feature(
+    feature: str, entry: Dict, value, tokenizer: Optional[Tokenizer]
+) -> Dict[str, Field]:
+    fields_to_be_added: Dict[str, Field] = dict()
+    if isinstance(value, ClassLabel):
+        fields_to_be_added[feature] = map_ClassLabel(feature, entry[feature])
+    # datasets Value can be of different types
+    elif isinstance(value, Value):
+        fields_to_be_added[feature] = map_Value(feature, entry[feature], value, tokenizer)
+
+    elif isinstance(value, Sequence):
+        fields_to_be_added = map_Sequence(feature, entry, value, tokenizer)
+
+    elif isinstance(value, Translation):
+        fields_to_be_added = map_Translation(feature, entry, value, tokenizer)
+
+    elif isinstance(value, TranslationVariableLanguages):
+        fields_to_be_added = map_TranslationVariableLanguages(feature, entry, value, tokenizer)
+
+    else:
+        raise ValueError(f"Datasets feature type {type(value)} is not supported yet.")
+    return fields_to_be_added
+
+
+def map_ClassLabel(feature: str, entry: Dict) -> Field:
+    field: Field = map_to_Label(feature, entry, skip_indexing=True)
+    return field
+
+
+def map_Value(
+    feature: str, item: Value, value, tokenizer: Optional[Tokenizer]
+) -> Union[TextField, LabelField]:
+    field: Union[TextField, LabelField]
+    if value.dtype == "string":
+        # datasets.Value[string] maps to TextField
+        # If tokenizer is provided we will use it to split it to tokens
+        # Else put whole text as a single token
+        field = map_String(feature, item, None, tokenizer)
+
+    else:
+        field = LabelField(item, label_namespace=feature, skip_indexing=True)
+    return field
+
+
+def map_Sequence(
+    feature: str, entry: Dict, value, tokenizer: Optional[Tokenizer]
+) -> Dict[str, Field]:
+    item_field: Union[LabelField, TextField]
+    field_list: List[Union[TextField, LabelField]] = list()
+    fields: Dict[str, Field] = dict()
+    if isinstance(value.feature, Value):
+        for item in entry[feature]:
+            # If tokenizer is provided we will use it to split it to tokens
+            # Else put whole text as a single token
+            item_field = map_Value(feature, item, value.feature, tokenizer)
+            field_list.append(item_field)
+        if len(field_list) > 0:
+            fields[feature] = ListField(field_list)
+
+    # datasets Sequence of strings to ListField of LabelField
+    elif isinstance(value.feature, ClassLabel):
+        for item in entry[feature]:
+            item_field = map_to_Label(feature, item, skip_indexing=True)
+            field_list.append(item_field)
+
+        if len(field_list) > 0:
+            fields[feature] = ListField(field_list)
+
+    else:
+        HuggingfaceDatasetReader.raise_feature_not_supported_value_error(value)
```

Comment:
I haven't looked at Huggingface datasets in enough detail, but I would have expected that there is some recursive call in here. What happens if one sequence feature contains another sequence feature?

Reply:
Looks like my claim appears to be wrong as of now. I had a dataset that would fail due to nesting and it passes giving me a false positive. 
I need to add recursion inside `_map_Sequence` with either `_map_Feature` or itself to handle that nesting. 
I will try to formulate that. 

Also I will try to find a dataset which has that so I can make sure it works.

Reply:
I have added the required recursion, code looks alright but haven't been able to validate. 
Take a look at the code and let me know if anything wrong pops
I will try to random sample the HF datasets in hub and see if any use sequence of sequence

Looks like sequence of dict is a popular case from a few datasets that start with a. 
I will try to Implement that now.